gadget/install/partition.go

```diff
@@ -159,18 +160,20 @@ func buildPartitionList(dl *gadget.OnDiskVolume, pv *gadget.LaidOutVolume) (sfdi
 
 		ptype := partitionType(dl.Schema, p.Type)
 
-		// Can we use the index here? Get the largest existing partition number and
-		// build from there could be safer if the disk partitions are not consecutive
-		// (can this actually happen in our images?)
+		// synthesize the node name and on disk structure
 		node := deviceName(dl.Device, pIndex)
-		fmt.Fprintf(buf, "%s : start=%12d, size=%12d, type=%s, name=%q\n", node,
-			startInSectors, newSizeInSectors, ptype, s.Name)
-
-		toBeCreated = append(toBeCreated, gadget.OnDiskStructure{
+		ps := gadget.OnDiskStructure{
 			LaidOutStructure: p,
 			Node:             node,
+			DiskIndex:        pIndex,
 			Size:             quantity.Size(newSizeInSectors * sectorSize),
-		})
+		}
+
+		// format sfdisk output for creating this partition
```

Comment:
s/output/input/

Reply:
Good catch, I will do this in the followup